packages/shared-components/src/components/onto-header/onto-header.tsx

```diff
@@ -1,21 +1,53 @@
-import { Component, Host, h } from '@stencil/core';
+import { Component, Host, h, State } from '@stencil/core';
+import { ServiceProvider, LicenseContextService, License } from '@ontotext/workbench-api'
 
+/**
+ * OntoHeader component for rendering the header of the application.
+ * This component includes a search component, license alert (if applicable),
+ * repository selector, and language selector.
+ */
 @Component({
   tag: 'onto-header',
   styleUrl: 'onto-header.scss',
   shadow: false,
 })
 export class OntoHeader {
+  /** The current license information */
+  @State() private license: License;
+
+  /** Array of subscription cleanup functions */
+  private readonly subscriptions: (() => void)[] = [];
```

Comment:
I think it'd be better if we have an explicit type for the subscription, e.g. `type Subscription = () => void`. Also, a SubscriptionList class with methods to add a subscription and unsubscribe all subscriptions. It'd be neat to have these I think.

Reply:
created new type `subscription` and class `subscription-list`. Added methods for adding a subscription and unsubscribing from all subscriptions